modules/aruco/test/test_charucodetection.cpp

```diff
@@ -677,4 +677,38 @@ TEST(Charuco, testCharucoCornersCollinear_false)
     EXPECT_FALSE(result);
 }
 
+TEST(Charuco, issue_14014)
+{
+    Ptr<aruco::Dictionary> dico = aruco::getPredefinedDictionary(aruco::PREDEFINED_DICTIONARY_NAME(cv::aruco::DICT_7X7_250));
+    Ptr<aruco::CharucoBoard> board = aruco::CharucoBoard::create(8, 5, 0.03455f, 0.02164f, dico);
+    Ptr<aruco::DetectorParameters> detectorParams = aruco::DetectorParameters::create();
+    detectorParams.get()->cornerRefinementMethod = aruco::CORNER_REFINE_SUBPIX;
+    detectorParams.get()->cornerRefinementMinAccuracy = 0.01;
+
+    std::string testImagePath = cvtest::TS::ptr()->get_data_path() + "aruco/";
+    Mat img = imread(testImagePath + "nonrecover.png");
+    vector<vector<Point2f> > rejectedPoints;
+    vector<Mat> corners;
+    vector<int> ids;
+
+    aruco::detectMarkers(img, dico, corners, ids, detectorParams, rejectedPoints);
+    ASSERT_FALSE(corners.empty());
+    EXPECT_EQ(Size(4, 1), corners[0].size());
```

Comment:
> `EXPECT_EQ(Size(4, 1), corners[0].size());`

Why do you think that `Size(1, 4)` is wrong?

Reply:
Size(1, 4) is not wrong (because it's just 4 2d-points: x1, y1, x2, y2, x3, y3, x4, y4) , but `corners[i]` must be the same size in all cases to avoid undefined behavior.=
I also think that `InputOutputArrayOfArrays` should not be used to pass array of pointers. You suggested replacing the `InputOutputArrayOfArrays` with a `InputOutputArray` with cv::Mat with Size(4, npoints) of CV_32FC2. I'll try to do that.

Reply:
Let me clarify:
In all tests and all examples in OpenCV detectMarkers return `corners` with `Size(4, 1)` (for `corners[i]`). It is strange if this condition is violated for no reason.